jax_md/smap.py

```diff
@@ -155,6 +251,9 @@ def pairwise(
     The mapped function can also optionally take keyword arguments that get
     threaded through the metric.
   """
+
+  metric = vmap(vmap(metric, (0, None), 0), (None, 0), 0) 
```

Comment:
For people looking at this code for the first time, or trying to write custom functionality and using this as a guide, this line is very cryptic. Perhaps a sentence or two that explains what is going on here? Or is that best left for when documentation is developed?